packages/typespec-client-generator-core/test/examples/string-conversion.test.ts

```diff
@@ -0,0 +1,295 @@
+import { expectDiagnostics } from "@typespec/compiler/testing";
+import { ok, strictEqual } from "assert";
+import { beforeEach, describe, it } from "vitest";
+import { SdkHttpOperation, SdkServiceMethod } from "../../src/interfaces.js";
+import { SdkTestRunner, createSdkTestRunner } from "../test-host.js";
+
```

Comment:
please move all these tests into types.test.ts and follow the structures and naming convention in that file.